Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/HttpContextExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿namespace Amazon.Lambda.TestTool;
+
+using Amazon.Lambda.APIGatewayEvents;
+using System.Text;
+using System.Web;
+using static Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyRequest;
+
+/// <summary>
+/// Provides extension methods to translate an <see cref="HttpContext"/> to different types of API Gateway requests.
+/// </summary>
+public static class HttpContextExtensions
+{
+    private static IHttpRequestUtility _httpRequestUtility = new HttpRequestUtility();
+    private static IRouteConfigurationParser _routeConfigurationParser;
+
+    public static void SetHttpRequestUtility(IHttpRequestUtility httpRequestUtility)
+    {
+        _httpRequestUtility = httpRequestUtility ?? throw new ArgumentNullException(nameof(httpRequestUtility));
+    }
+
+    public static void SetRouteConfigurationParser(IRouteConfigurationParser routeConfigurationParser)
+    {
+        _routeConfigurationParser = routeConfigurationParser ?? throw new ArgumentNullException(nameof(routeConfigurationParser));
+    }
+
+    /// <summary>
+    /// Translates an <see cref="HttpContext"/> to an <see cref="APIGatewayHttpApiV2ProxyRequest"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="HttpContext"/> to be translated.</param>
+    /// <returns>An <see cref="APIGatewayHttpApiV2ProxyRequest"/> object representing the translated request.</returns>
+    public static APIGatewayHttpApiV2ProxyRequest ToApiGatewayHttpV2Request(
+        this HttpContext context)
+    {
+        var request = context.Request;
+
+        var matchedConfig = _routeConfigurationParser.GetRouteConfig(request.Method, request.Path);
```

Comment:
this should be removed from here or put in some other level of abstraction. will sync with phil today